src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -312,15 +312,75 @@ internal static bool IsCentralPackageTransitivePinningEnabled(IReadOnlyList<IVsT
             string? enableAudit = GetSingleNonEvaluatedPropertyOrNull(tfms, ProjectBuildProperties.NuGetAudit, s => s);
             string? auditLevel = GetSingleNonEvaluatedPropertyOrNull(tfms, ProjectBuildProperties.NuGetAuditLevel, s => s);
             string? auditMode = GetSingleNonEvaluatedPropertyOrNull(tfms, ProjectBuildProperties.NuGetAuditMode, s => s);
+            HashSet<string>? suppressedAdvisories = GetSuppressedAdvisories(tfms);
 
-            return !string.IsNullOrEmpty(enableAudit) || !string.IsNullOrEmpty(auditLevel) || !string.IsNullOrEmpty(auditMode)
+            return !string.IsNullOrEmpty(enableAudit) || !string.IsNullOrEmpty(auditLevel) || !string.IsNullOrEmpty(auditMode) || suppressedAdvisories is not null
                 ? new RestoreAuditProperties()
                 {
                     EnableAudit = enableAudit,
                     AuditLevel = auditLevel,
                     AuditMode = auditMode,
+                    SuppressedAdvisories = suppressedAdvisories,
                 }
                 : null;
+
+            static HashSet<string>? GetSuppressedAdvisories(IReadOnlyList<IVsTargetFrameworkInfo4> tfms)
+            {
+                if (tfms.Count == 0) { return null; }
+
+                // Create the hash set from the first TargetFramework
+                HashSet<string>? suppressedAdvisories = null;
+                if (tfms[0].Items?.TryGetValue(ProjectItems.NuGetAuditSuppress, out IReadOnlyList<IVsReferenceItem2>? suppressItems) ?? false)
+                {
+                    suppressedAdvisories = new(suppressItems.Count, StringComparer.Ordinal);
+                    for (int i = 0; i < suppressItems.Count; i++)
+                    {
+                        string url = suppressItems[i].Name;
+                        suppressedAdvisories.Add(url);
+                    }
+                }
+
+                // Validate that other TargetFrameworks use the same collection
+                for (int i = 1; i < tfms.Count; i++)
+                {
+                    if (!AreSameAdvisories(suppressedAdvisories, tfms[i].Items))
+                    {
+                        string message = string.Format(Resources.ItemValuesAreDifferentAcrossTargetFrameworks, ProjectItems.NuGetAuditSuppress);
+                        throw new InvalidOperationException(message);
+                    }
+                }
+
+                return suppressedAdvisories;
+
+                static bool AreSameAdvisories(HashSet<string>? suppressedAdvisories, IReadOnlyDictionary<string, IReadOnlyList<IVsReferenceItem2>>? items)
```

Comment:
Are you better off using some of the EqualityUtilities we have?

Reply:
I forgot about those methods. However, in order to use it, we'd have to create a new `HashSet<string>` for each TFM. The way this is currently written causes zero allocations after the first TFM. Therefore, whether we're better off depends on your opinion of code reuse vs memory allocations.